src/adagrams.js

```diff
@@ -1,15 +1,116 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = 
+    {A:9, B:2, C:2, D:4, E:12, F:2, G:3, 
+    H:2, I:9, J:1, K:1, L:4, M:2, N:6, O:8, P:2, 
+    Q:1, R:6, S:4, T:6, U:4, V:2, W:2, X:1, Y:2, Z:1
+  };
+
+  const computeCumulativeWeights = (letterPool) => {
+    const cumulativeWeights = {};
+    let totalWeight = 0;
+
+    for (const [letter, weight] of Object.entries(letterPool)) {
+      totalWeight += weight;
+      cumulativeWeights[letter] = totalWeight;
+    }
+
+    return {cumulativeWeights, totalWeight};
+  };
+
+  const letterPoolCopy = {...letterPool};
+  const selectedLetters = [];
+  
+  while (selectedLetters.length < 10) {
+  const {cumulativeWeights, totalWeight} = computeCumulativeWeights(letterPoolCopy);
+  const randomValue = Math.floor(Math.random() * totalWeight + 1);
+    
+  for (const [letter, weight] of Object.entries(cumulativeWeights)) {
+    if (randomValue <= weight) {
+      selectedLetters.push(letter);
+      
+      letterPoolCopy[letter] = letterPoolCopy[letter] - 1;
+      if (letterPoolCopy[letter] === 0) {
+        delete letterPoolCopy[letter];
+      }
+
+        break;
```

Comment:
Just some formatting
```suggestion
  for (const [letter, weight] of Object.entries(cumulativeWeights)) {
    if (randomValue <= weight) {
      selectedLetters.push(letter);
      letterPoolCopy[letter] = letterPoolCopy[letter] - 1;
      
      if (letterPoolCopy[letter] === 0) {
        delete letterPoolCopy[letter];
      }

        break;
```